tests/nat/builder/test_per_user_builder.py

```
@@ -0,0 +1,1225 @@
+async def test_per_user_builder_builds_per_user_function_groups():
+    """Test that PerUserWorkflowBuilder builds per-user function groups."""
+    import typing
+
+    from nat.builder.function import FunctionGroup
+    from nat.cli.register_workflow import register_per_user_function_group
+
+    class TestPerUserFGConfig(FunctionGroupBaseConfig, name="test_pu_fg_build"):
+        prefix: str
+
+    @register_per_user_function_group(config_type=TestPerUserFGConfig)
+    async def test_per_user_fg(config: TestPerUserFGConfig, builder: Builder):
```

Ruff (0.14.8)

Unused function argument: `builder`

(ARG001)